go/http/gen_sign_str.go

```diff
@@ -83,17 +85,28 @@ func GetCanonicalRequest(req *http.Request) string {
 	return canonicalRequest
 }
 
+// Deprecated: This method will be deleted in future versions, once most SP and clients migrates to GNFD1 Auth.
+// See GetMsgToSignInGNFD1
 // GetMsgToSign generate the msg bytes from canonicalRequest to sign
 func GetMsgToSign(req *http.Request) []byte {
+	signBytes := hash.GenerateChecksum([]byte(GetCanonicalRequest(req)))
+	return crypto.Keccak256(signBytes)
+}
+
+// GetMsgToSignInGNFD1Auth generate the msg bytes from canonicalRequest to sign
+// This method will be used for the following GNFD1 Auth algorithms:
+// - GNFD1-ECDSA
+// - GNFD1-EDDSA
+func GetMsgToSignInGNFD1Auth(req *http.Request) []byte {
```

Comment:
The naming is a bit confusing，if we need to support different types by this function,  we can consider add a type parameter like GetMsgToSign(req *http.Request, type *SignType) .  and define the  const SignType with GNFD1-ECDSA or GNFD1-EDDSA,  and we can support more if we have more type like GNFD2-EDDSA